src/server/services/registrar.js

```diff
@@ -164,8 +164,61 @@ async function getShortname(offeredTermCode, classSectionID) {
   }
 }
 
+/**
+ * Gets the week number, given an academic term and a date
+ *
+ * @param {string} term Academic term, formatted as YYQ (e.g. 20S)
+ * @param {string} date Date string formatted as MM/DD/YYYY
+ * @returns {?string} Returns week number if date is valid in term
+ */
+async function getWeekNumber(term, date) {
+  try {
+    const response = await registrar.call({
+      url: `sis/api/v1/Dictionary/TermSessionsByWeek?SessionTermCode=${term}&SessionCode=RG&PageSize=12`,
```

Comment:
@chern If I am reading this right you are making a web service call for every Bruincast entry, even though the term details don't change right?

You should cache the response and have the function check that cache first before making a call request.

I did a quick search on "node cache" and there's lots of options. Please figure out the best general caching solution. I am sure we will be wanting to cache more data later on when we optimize the application.

* https://scotch.io/tutorials/how-to-optimize-node-requests-with-simple-caching-strategies
* https://www.mojotech.com/blog/node-js-memory-cache/
* https://www.npmjs.com/package/node-cache

Reply:
@chern Should cache the processed week data. I see in the loop you are calling a split operation and getting the startDate and endDate objects over and over again. That should be cached as a processed term, so we only do that operation once.